cmake/compileroptions.cmake

```diff
@@ -32,7 +32,7 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
     add_compile_options(-Wall)
     add_compile_options(-Wextra)
     add_compile_options(-Wcast-qual)                # Cast for removing type qualifiers
-    add_compile_options(-Wfloat-equal)              # Floating values used in equality comparisons
+    # add_compile_options(-Wfloat-equal)              # Floating values used in equality comparisons
```

Comment:
This is also enable in the Makefile by default. If we no longer care about it we should just remove it everywhere (and disable it explicitly for Clang since we enable all warnings for it).

Reply:
In general I would agree with the warning but I believe there are some case in cppcheck code that we don't want to warn about.